crates/store/re_video/src/lib.rs

```diff
@@ -22,6 +21,43 @@ pub struct VideoData {
     pub data: Vec<u8>,
 }
 
+impl VideoData {
+    /// Loads a video from the given data.
+    ///
+    /// TODO(andreas, jan): This should not copy the data, but instead store slices into a shared buffer.
+    /// at the very least the should be a way to extract only metadata.
+    pub fn load_from_bytes(data: &[u8], media_type: Option<&str>) -> Result<Self, VideoLoadError> {
+        // Media type guessing here should be identical to `re_types::MediaType::guess_from_data`,
+        // but we don't want to depend on `re_types` here.
+        let media_type = if let Some(media_type) = media_type {
+            media_type.to_owned()
+        } else if mp4::is_mp4(data) {
+            "video/mp4".to_owned()
+        } else {
+            // Technically this means that we failed to determine the media type altogether,
+            // but we don't want to call it `FailedToDetermineMediaType` since the rest of Rerun has
+            // access to `re_types::components::MediaType` which has a much wider range of media type detection.
+            return Err(VideoLoadError::UnsupportedVideoType);
+        };
+
+        match media_type.as_str() {
+            "video/mp4" => mp4::load_mp4(data),
+            media_type => Err(VideoLoadError::UnsupportedMediaType(media_type.to_owned())),
+        }
+    }
+
+    /// Determines the presentation timestamps of all frames inside a video, returning raw time values.
+    ///
+    /// Returned timestamps are in nanoseconds since start and are guaranteed to be monotonically increasing.
```

Comment:
I don't see anything enforcing this guarantee. At a minimum I would expect a check for the condition and an error pathway inside of `mp4::load_mp4`. Isn't this sort of what we ran into with h264?

Reply:
hmm now that you mention it, I do think there might be an issue still here. AVC doesn't ever have problems with this, but the mp4 container will have this issue once we properly support h264 since the better order for the container is decode frame order rather than presentation frame order!